### ISSUE TITLE
Please add Vim Mode！！！

### ISSUE THREAD

**Issue by LaoYang1994** [cpeditor/cpeditor#220]
It's such a wonderful editor. Could you add vim mode in it?

**Comment by ouuan** [comment]
We have many other plans now, and vim mode is not easy to implement. If there are more people asking for this, we may increase the priority.

**Comment by LaoYang1994** [comment]
@ouuan Thanks.

**Comment by coder3101** [comment]
I have started the work on adding vim mode to QCodeEditor at [this branch](https://github.com/cpeditor/QCodeEditor/tree/fakevim)

You can expect initial support for vim after next few releases, we will let you know for testing when it hits alpha/beta

**Comment by coder3101** [comment]
Development for vim emulation is Code Editor independent. I am actively working on adding this support as I like vim binding and use it everywhere!

You can track the progress on branch named `fakevim-support`.

**Comment by coder3101** [comment]
Vim support is mostly complete. We are just left with adding some custom vim command supports so you can run/compile directly by issuing a vim command. We might need some of you to test it when we are done.

**Comment by tuwuna** [comment]
Yay, its almost here :D

**Comment by ForgotDream** [comment]
Any progress? I really wish to code like vim on such a great editor!

**Comment by coder3101** [comment]
TLDR;

> The PR which proposed to add this has been closed.


#516 was to add ksyntax highlighter + vim emulation.

After this PR became stale, #1101 was created, inspired from 516 but excluding vim emulation. This has been merged!

Someone can open a vim emulation PR inspired from 516 and we will be happy to review and merge it.

**Comment by coder3101** [comment]
@rogeryoungh are you willing to port Vim Emulation from #516

**Comment by rogeryoungh** [comment]
Sorry, I've been preparing for an important exam lately and won't be available for a few months.

**Comment by coder3101** [comment]
Try out vim mode by building from PR #1270 
Do report any bugs, ideas or improvements.
